.github/workflows/check.yml

```diff
@@ -1,43 +1,87 @@
 name: Check
 
-# XXX disabled for now because we use Jenkins still, but at the time this was tested it was functional
-# on:
-#   pull_request:
-#     branches: [ master ]
+on:
+  pull_request:
+    branches:
+      - '*'  # Trigger on all branches for pull requests
+  push:
+    branches:
+      - master  # Trigger on pushes to the master branch
 
 jobs:
-  check:
+  check_pull_request:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v3
+      - name: Checkout code
+        uses: actions/checkout@v3
 
-    # Note: Actually we don't need Maven CLI, only Java 17, but this may have additional benefits due to the Maven cache
-    - name: Setup Maven
-      uses: s4u/setup-maven-action@v1.5.1
-      with:
-        java-version: 17
-        java-distribution: temurin
-        maven-version: 3.8.6
+      - name: Setup Maven
+        uses: s4u/setup-maven-action@v1.5.1
+        with:
+          java-version: 17
+          java-distribution: temurin
+          maven-version: 3.8.6
+          notify-failure: false
+          skip-build: false
+          expect-tests: false
+          submodules: 'recursive'
```

Comment:
I don't see the input parameters you added as supported parameters for the [action](https://github.com/marketplace/actions/setup-maven-action).